diff --git a/libs/librepcb/ui/CMakeLists.txt b/libs/librepcb/ui/CMakeLists.txt
index 4281b6867..fc4202949 100644
--- a/libs/librepcb/ui/CMakeLists.txt
+++ b/libs/librepcb/ui/CMakeLists.txt
@@ -15,7 +15,7 @@ slint_target_sources(
   # Generate multiple cpp files to parallelize compilation, but not too many
   # because of excessive RAM usage when compiling those (huge) files. A value
   # of 12 already caused a compiler segfault on the macOS CI...
-  COMPILATION_UNITS 8
+  COMPILATION_UNITS 1
 )
 target_link_libraries(
   librepcb_ui
